src/solvers/dg_multi/fluxdiff.jl

```diff
@@ -0,0 +1,182 @@
+
+  rd = dg.basis
+  @unpack md = mesh
+
+  # Todo: simplices. Fix this when StartUpDG v0.11.0 releases: new API `Qrst_hybridized, VhP, Ph = StartUpDG.hybridized_SBP_operators(rd)`
+  if Dim == 2
+    Qr_hybridized, Qs_hybridized, VhP, Ph = StartUpDG.hybridized_SBP_operators(rd)
+    Qrst_hybridized = (Qr_hybridized, Qs_hybridized)
+  elseif Dim == 3
+    Qr_hybridized, Qs_hybridized, Qt_hybridized, VhP, Ph = StartUpDG.hybridized_SBP_operators(rd)
+    Qrst_hybridized = (Qr_hybridized, Qs_hybridized, Qt_hybridized)
+  end
+  Qrst_skew_Tr = map(A -> -.5*(A-A'), Qrst_hybridized)
+
+
+  nvars = nvariables(equations)
+
+  # storage for all quadrature points (concatenated volume / face quadrature points)
+  num_quad_points_total = rd.Nq + rd.Nfq
+  entropy_projected_u_values = StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(num_quad_points_total, md.num_elements), nvars))
+  projected_entropy_var_values = similar(entropy_projected_u_values)
+
+  # initialize views into entropy_projected_u_values
+  u_values = view(entropy_projected_u_values, 1:rd.Nq, :)
+  u_face_values = view(entropy_projected_u_values, rd.Nq+1:num_quad_points_total, :)
+
+  # temp storage for entropy variables at volume quad points
+  entropy_var_values = StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(rd.Nq, md.num_elements), nvars))
+
+  # local storage for interface fluxes, rhs, and source
+  flux_face_values = similar(u_face_values)
+  rhs_local_threaded = [StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(num_quad_points_total), nvars)) for _ in 1:Threads.nthreads()]
+  local_values_threaded = [StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(rd.Nq), nvars)) for _ in 1:Threads.nthreads()]
+
+  return (; md, Qrst_skew_Tr, VhP, Ph, invJ = inv.(md.J),
+      entropy_var_values, projected_entropy_var_values, entropy_projected_u_values,
+      u_values, u_face_values, rhs_local_threaded, flux_face_values, local_values_threaded)
+end
+
+function entropy_projection!(cache, u::StructArray, mesh::VertexMappedMesh,
+                             equations, dg::DG) where {DG <: PolyDGFluxDiff}
+
+  rd = dg.basis
+  @unpack Vq = rd
+  @unpack VhP, entropy_var_values, u_values, entropy_var_values = cache
+  @unpack projected_entropy_var_values, entropy_projected_u_values = cache
+
+  # TODO: simplices. Address hard-coding of `entropy2cons!`
+  StructArrays.foreachfield(mul_by!(Vq), u_values, u)
+  entropy_var_values .= cons2entropy.(u_values, equations)
+
+  # "VhP" fuses the projection "P" with interpolation to volume and face quadrature "Vh"
+  StructArrays.foreachfield(mul_by!(VhP), projected_entropy_var_values, entropy_var_values)
+  entropy_projected_u_values .= entropy2cons.(projected_entropy_var_values, equations)
+end
+
+function calc_volume_integral!(du, u::StructArray, volume_integral::VolumeIntegralFluxDifferencing,
+                               mesh::VertexMappedMesh, equations, dg::PolyDGFluxDiff, cache)
```

Comment:
```suggestion
function calc_volume_integral!(du, u::StructArray, volume_integral,
                               mesh::VertexMappedMesh, equations, dg::PolyDGFluxDiff, cache)
```
maybe unnecessary (same question as above)?